src/core/config/mac.h

```diff
@@ -254,4 +255,26 @@
 #define OPENTHREAD_CONFIG_MAC_RETX_POLL_PERIOD 1000
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_CSL_TRANSMITTER_ENABLE
+ *
+ * This setting configures csl transmit feature in Thread 1.2.
+ * This is compulsory for 1.2 FTD.
+ *
+ */
+#define OPENTHREAD_CONFIG_CSL_TRANSMITTER_ENABLE \
```

Comment:
I had a discussion with @librasungirl that, since CSL Transmitter feature has to be enabled for a 1.2 Thread FTD device, we can always turn it on when it's 1.2 Thread FTD. However for CSL Receiver feature, if the device is a FTD, it's very unlikely that it would be used as an SSED, so we remain this as optional. Thoughts?

Reply:
SGTM.